app/Http/Controllers/Home/HomeController.php

```diff
@@ -12,17 +14,24 @@ class HomeController extends Controller
 {
     public function index(Request $request): View
     {
-        $activities = StravaActivity::byUser(auth()->user())
-            ->with('dswAnalysis.dswType.typeGroup')
-            ->latest('started_at')
-            ->paginate(20);
+        $activities = app(HomeFilteringService::class)->applyFiltersAndSort(
+            StravaActivity::byUser(auth()->user())->with('dswAnalysis.dswType.typeGroup')->latest('started_at'),
+            $request->all()['filters'] ?? []
```

Comment:
We need to verify that this is an array, and then can use a method like `->get('filters')` rather than extracting the value from `$request->all()`